### PR TITLE
MDEV-17654: Prepend COMMENT before CHECK constraint in SHOW CREATE TABLE

### PR DESCRIPTION
**Pull request by an3l**

### CODE REVIEW

**Comment by grooverdan** [comment]
Append -> Prepend. Otherwise looks fine.

**Comment by an3l** [comment]
Thanks @grooverdan

**Comment by JonathanWilbur** [comment]
@grooverdan Do you know when this will be merged? I am blocking a project solely on this. Thanks in advance!

**Comment by grooverdan** [comment]
Sorry @JonathanWilbur, I've no idea or any control over the acceptance/merging of submissions.

https://jira.mariadb.org/browse/MDEV-17654 just got a priority bump to Critical 8 hrs ago. So likely soon.

File: sql/sql_show.cc [hunk 1]
```diff
@@ -2031,11 +2038,6 @@ int show_create_table(THD *thd, TABLE_LIST *table_list, String *packet,
       packet->append(STRING_WITH_LEN(")"));
     }
 
-    if (field->comment.length)
-    {
-      packet->append(STRING_WITH_LEN(" COMMENT "));
-      append_unescaped(packet, field->comment.str, field->comment.length);
-    }
     append_create_options(thd, packet, field->option_list, check_options,
                           hton->field_options);
```
Comment (vuvova) [inline sql/sql_show.cc]: `append_create_options` should also be before the constraint. And please add a test case for it.

Comment (an3l) [reply sql/sql_show.cc]: @vuvova I have updated.

Comment (vuvova) [reply sql/sql_show.cc]: I don't understand your new test case, sorry.
Are you testing whether create options are shows in `SHOW CREATE TABLE`?
There was no need to. First, this is already tested elsewhere. And second — you did not change it, so you are not testing your changes.

You've changed the order in which create options are printed as compare to `CHECK` constraint. This is what you should test. For example,
```
create table t1 (f int foo=bar check(f>0));
show create table t1;
```

**Pull request closed without merging**